src/darker/formatters/black_formatter.py

```diff
@@ -96,6 +96,27 @@ def read_config(self, src: tuple[str, ...], args: Namespace) -> None:
         self._read_cli_args(args)
 
     def _read_config_file(self, config_path: str) -> None:  # noqa: C901
+        # Local import so Darker can be run without Black installed.
+        # Do error handling here. This is the first Black importing method being hit.
+        try:
+            from black import (  # pylint: disable=import-outside-toplevel
+                parse_pyproject_toml,
+                re_compile_maybe_verbose,
+            )
+        except ImportError as exc:
+            logger.warning(
+                "To re-format code using Black, install it using e.g."
+                " `pip install 'darker[black]'` or"
+                " `pip install black`"
+            )
+            logger.warning(
+                "To use a different formatter or no formatter, select it on the"
+                " command line (e.g. `--formatter=none`) or configuration"
+                " (e.g. `formatter=none`)"
+            )
+            message = "Can't find the Black package"
+            raise DependencyError(message) from exc
```

Comment:
What will happen, if the internals of black will change? e.g.: renaming/moving of `parse_pyproject_toml` or `re_compile_maybe_verbose` ? Think the error message is not really good in this case.

So, what's about to import only black package and then do the real imports? 

e.g.:
```
        try:
            import black
        except ImportError as exc:
            logger.warning(
                "To re-format code using Black, install it using e.g."
                " `pip install 'darker[black]'` or"
                " `pip install black`"
            )
            logger.warning(
                "To use a different formatter or no formatter, select it on the"
                " command line (e.g. `--formatter=none`) or configuration"
                " (e.g. `formatter=none`)"
            )
            message = "Can't find the Black package"
            raise DependencyError(message) from exc

        # Black is installed
        from black import (  # pylint: disable=import-outside-toplevel
            parse_pyproject_toml,
            re_compile_maybe_verbose,
        )
```